### ISSUE TITLE
Detecting Faces

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#8]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Detecting Faces </br>
:red_circle: **Aim** : The aim is to create a deep learning project which will detect the faces using OpenCV and MTCNN approach with an accuracy over 85%.</br>
:red_circle: **Dataset** : https://www.kaggle.com/code/wittmannf/detecting-faces-using-opencv-mtcnn-no-internet/data</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by AbhishekRP2002** [comment]
Hey there ! Can I work on this issue ? I would like to contribute to this .

**Comment by AbhishekRP2002** [comment]
Full name : Abhishek Ranjan Prusty
GitHub Profile Link : https://github.com/AbhishekRP2002
Email ID : aviranjan444@gmail.com
Participant ID (if applicable): 
Approach for this Project :  This model that we will use for Face detection will be  Multi-task Cascaded Convolutional Networks (MTCNN), which is essentially several convolutional networks strung together that give out several pieces of information. The neural network is expected to detect individual faces, and locate facial landmarks (i.e. two eyes, nose, and endpoints of the mouth), along with  drawing a bounding box around the face. We will use  OpenCV to open, read, write, and show images.
What is your participant role? (SSOC 2022)

**Comment by Sahana915** [comment]
Issue no: #8 Detecting Faces
Full Name : Sahana B Kanteppagoudra
Github profile link: https://github.com/Sahana915
Participation role : SSOC 2022
Approach: I will try out by using variety of DL algorithms like Haar Cascade or I'll go by using HOG based detector in dlib .

**Comment by Rushijaviya** [comment]
Assigned to @AbhishekRP2002. @Sahana915, you can choose other issues; we assign issues based on FCFS.

**Comment by abhisheks008** [comment]
I am sorry to say that @AbhishekRP2002 has already assigned an issue, complete that one first. This issue is assigned to @Sahana915.

@AbhishekRP2002 Do not comment out in other issues while you have already assigned to an issue. Finish that first, then take other issues.

**Comment by abhisheks008** [comment]
@Sahana915 updates please.

**Comment by abhisheks008** [comment]
Updates please @Sahana915

**Comment by Singhal-Akshat** [comment]
Full name: Akshat Singhal
GitHub Profile Link :
Email ID: akshatsinghal2105@gmail.com
Approach for this Project: I will use MTCNN and OpenCV, along with a comparison with the Haarcascade method, for MTCNN will use Dlib.
What is your participant role?  SSOC (2023)

**Comment by abhisheks008** [comment]
Issue assigned to you @Singhal-Akshat

**Comment by SaketGudimella** [comment]
Full name: Saket Gudimella
GitHub Profile Link : https://github.com/SaketGudimella
Email ID: radhsaket38@gmail.com
Approach for this Project: In this project, I will compare the performance of OpenCV's Haar cascades, MTCNN, Dlib, and RetinaFace algorithms to determine the most accurate and suitable approach for face detection.

What is your participant role? SSoC'23 Contributor 

@abhisheks008 Please assign me this issue under the SSoC'23 label.

**Comment by saikrishna823** [comment]
hi, @abhisheks008 ,could you please assign issue to me.
Full Name:Mule Sai Krishna Reddy
Github Profile Link:https://github.com/saikrishna823
Email ID:[20131a05f4@gvpce.ac.in](mailto:20131a05f4@gvpce.ac.in)
Participant ID (if applicable):
Approach for this Project : I will use MTCNN,Haar cascade opencv ,yolo,SSD too to detect faces in real time.
What is your participant role? I am participating as contributor through gssoc' 24.

**Comment by abhisheks008** [comment]
Hi @saikrishna823 thanks for showing up with your approach. If you can implement these algorithms this will be good project to include in this project repo. Stick to your approach and make sure you implement as many algorithms you can.

Assigned this issue to you @saikrishna823

**Comment by saikrishna823** [comment]
@abhisheks008 I am facing issues with downloading dataset.

**Comment by abhisheks008** [comment]
What are the issues you are facing?

**Comment by saikrishna823** [comment]
Not able to download dataset.It is under competiton.

**Comment by abhisheks008** [comment]
Ohhh! Can you find another dataset of same use from a trusted source (Kaggle will be better) for this project?

**Comment by saikrishna823** [comment]
okay

**Comment by Diyaa0313** [comment]
Full name : Diya Sen
GitHub Profile Link : https://github.com/Diyaa0313
Email ID : diyasen2003@gmail.com
Participant ID (if applicable): NA
Approach for this Project :

1. I will use EDA to plot the distribution of faces per image and use histograms to understand the variation in face sizes and orientation and remove any imbalance with oversampling or undersampling.

2. Use MTCNN, SSD, and YOLO for face detection and optimize the hyperparameters.

3. Compare which model gives higher accuracy.

4. Run the model on the image to detect faces. Draw bounding boxes around the detected faces using OpenCV.

What is your participant role? GSSOC 2024 Contributor

**Comment by abhisheks008** [comment]
> Full name : Diya Sen GitHub Profile Link : https://github.com/Diyaa0313 Email ID : [diyasen2003@gmail.com](mailto:diyasen2003@gmail.com) Participant ID (if applicable): NA Approach for this Project :
> 
> 1. I will use EDA to plot the distribution of faces per image and use histograms to understand the variation in face sizes and orientation and remove any imbalance with oversampling or undersampling.
> 2. Use MTCNN, SSD, and YOLO for face detection and optimize the hyperparameters.
> 3. Compare which model gives higher accuracy.
> 4. Run the model on the image to detect faces. Draw bounding boxes around the detected faces using OpenCV.
> 
> What is your participant role? GSSOC 2024 Contributor

This issue is already assigned to a contributor. You can check out the issues which are having `Up for Grabs` tag/label,

**Comment by saikrishna823** [comment]
Hi, @abhisheks008 I found image dataset.If it is okay for you,I will start implementing the project

**Comment by abhisheks008** [comment]
> Hi, @abhisheks008 I found image dataset.If it is okay for you,I will start implementing the project

Can you please share the source/URL of the dataset before start working on it?
@saikrishna823

**Comment by saikrishna823** [comment]
Hi @abhisheks008 ,I am sharing the url of the dataset http://shuoyang1213.me/WIDERFACE/.Have a look at it.

**Comment by abhisheks008** [comment]
Cool you can go ahead with this issue. Assigning this issue to you @saikrishna823

**Comment by Raghucharan16** [comment]
> Hi @abhisheks008 ,I am sharing the url of the dataset http://shuoyang1213.me/WIDERFACE/

hey @saikrishna823 nice to see you here, you have already developed your own algo right??

**Comment by why-Radhika** [comment]
I noticed it has been more than a month since the issue has been assigned and there is no PR for the same. So, I would like to work on this issue.

Full name : Radhika Gupta
GitHub Profile Link : https://github.com/why-Radhika
Email ID : gupta.rads2003@gmail.com
Approach for this Project : Multi-task Cascaded Convolutional Networks (MTCNN). MTCNN consists of multiple convolutional networks arranged in a cascaded manner, designed to output several pieces of information simultaneously. Specifically, it will detect individual faces within images, accurately locate facial landmarks such as the eyes, nose, and endpoints of the mouth, and draw bounding boxes around detected faces. In conjunction with MTCNN, we will leverage OpenCV for tasks such as image manipulation, reading and writing images, and visualizing the detection results.
What is your participant role? (Mention the Open Source program) : GSSOC'24

**Comment by abhisheks008** [comment]
Hi @why-Radhika go ahead with the issue.

Unassigning this issue @saikrishna823 due to inactivity.

**Comment by that-ar-guy** [comment]
HEY @abhisheks008 can i take this up under SWOC

**Comment by abhisheks008** [comment]
> HEY [@abhisheks008](https://github.com/abhisheks008) can i take this up under SWOC

Yes sure!!

Please mention you approach for solving this problem statement.

**Comment by that-ar-guy** [comment]
@abhisheks008 , 
So my approach will focus on building a simple face detection system using the Haar Cascade classifier. Haar Cascades are lightweight and effective for real-time face detection, making them suitable for this task.

To implement the project, I plan to integrate the Haar Cascade method with a webcam feed. Using OpenCV, I will utilize VideoCapture to capture live video and imshow for real-time visualization of detected faces. The system will identify and draw bounding boxes around faces detected in the webcam feed, demonstrating its functionality in an interactive manner.

**Comment by abhisheks008** [comment]
You can start working on this issue. Assigned to you @that-ar-guy
